app/facades/revenue_facade.rb

```diff
@@ -4,4 +4,8 @@ def self.merchant_revenue(merchant)
 
     Revenue.new(result)
   end
+
+  def self.merchants_by_revenue(quantity)
+    Merchant.joins(invoices: [:invoice_items, :payments]).select("merchants.*, sum(invoice_items.quantity * invoice_items.unit_price) as revenue").where("payments.result='success' AND invoices.status='shipped'").group(:id).order("revenue desc").limit(quantity)
```

Comment:
Another tricky AR query.Creating relationships in the merchant model was key for getting this done.